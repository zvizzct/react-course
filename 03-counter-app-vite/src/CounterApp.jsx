import { useState } from 'react'

export const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value);

    const handleAdd = () => {
        setCounter(counter + 1);
    }
    const handleReset = () => {
        setCounter(value);
    }

    const handleSub = () => {
        setCounter(counter - 1);
    }

    return (
        <>
            <h1> CounterApp </h1>
            <h2>{counter}</h2>
            <div className="butt">
                <button onClick={handleSub}> -1 </button>
                <button aria-label="btn-reset"  onClick={handleReset}> Reset </button>
                <button onClick={handleAdd}> +1 </button>
            </div>
        </>
    )


}