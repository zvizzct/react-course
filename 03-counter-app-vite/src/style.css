body,
html {
    margin: 0;
    padding: 20px;
    box-sizing: border-box;
    font-family: 'Roboto', sans-serif;
    background-color: #343232;
}

h1,
h2,
p {
    color: #fff;
}

.butt {
    display: flex;
    gap: 5px
}

button {
    background-color: rgb(74, 73, 73);
    border: none;
    border-radius: 5px;
    padding: 10px;
    font-size: 1.2rem;
    cursor: pointer;
    border: 1px solid #fff;
    color: #fff;
    transition: all 0.12s ease-in-out;
}

button:hover {
    background-color: #fff;
    color: #000;
}