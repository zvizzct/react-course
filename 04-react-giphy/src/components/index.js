export * from './AddCategory';
export * from './GifGrid';
export * from './GifItem';
export * from './Header';
export * from './GifHistory';
