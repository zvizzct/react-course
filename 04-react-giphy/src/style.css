/* General styles */

* {
  font-family: ui-sans-serif, system-ui, -apple-system, BlinkMacSystemFont,
    Segoe UI, Roboto, Helvetica Neue, Arial, Noto Sans, sans-serif,
    'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
}
/* Theme styles */
.light-theme {
  --background-color: #ffffff;
  --text-color: #000000;
}

.dark-theme {
  --background-color: #2c2c2c;
  --text-color: #ffffff;
}

/* Body styles */
body {
  padding: 10px 60px;
  background-color: var(--background-color);
  transition: background-color 0.3s ease, color 0.3s ease;
}

/* Container styles */
.container {
  max-width: 1200px;
  margin: 0 auto;
}

/* Header styles */
header {
  display: flex;
  align-items: center;
  justify-content: space-between;
  overflow: hidden;
  margin-top: -30px;
}

header h1 {
  background-image: linear-gradient(
    to left,
    rgb(127, 234, 77),
    rgb(115, 208, 237),
    rgb(254, 78, 242),
    rgb(242, 94, 57),
    rgb(248, 235, 86)
  );
  font-size: 3rem;
  -webkit-background-clip: text;
  color: transparent;
  background-size: 200% 200%;
  animation: moving-gradient 2s linear infinite;
}

header p {
  font-size: 3rem;
}

/* Form styles */
form {
  display: flex;
  justify-content: center;
}

input {
  background-color: white;
  border-radius: 5px;
  border: 3px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 50%;
  transition: all 0.2s ease;
}

input:focus {
  background-color: rgb(220, 220, 220);
}

/* History section styles */
.section-history {
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 70px;
  gap: 10px;
}

.btn-history {
  background-color: rgba(177, 173, 187, 0.823);
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  font-weight: bold;
  color: black;
  font-size: 0.8rem;
  outline: none;
  padding: 10px 15px;
  cursor: pointer;
}

/* Typography styles */
h2 {
  font-size: 1.5rem;
  color: var(--text-color);
}

h3 {
  font-size: 3rem;
  margin-bottom: 5px;
  color: var(--text-color);
}

/* Card styles */
.card {
  align-content: center;
  align-items: center;
  border-radius: 10px;
  cursor: pointer;
  width: 380px;
  overflow: hidden;
  transition: all 0.2s ease;
}

.card img {
  height: 250px;
  object-fit: cover;
  width: 100%;
  border-radius: 10px;
  border: 1px solid rgb(97, 32, 158);
  transition: all 0.2s ease;
}
/* Card grid styles */
.card-grid {
  display: flex;
  flex-wrap: wrap;
  justify-content: stretch;
  gap: 20px;
}

.card:hover {
  scale: 1.1;
  filter: brightness(0.7);
}

.card img:hover {
  scale: 1.8;
}

/* Keyframe animation */
@keyframes moving-gradient {
  0% {
    background-position: 0% 100%;
  }
  50% {
    background-position: 100% 50%;
  }
  100% {
    background-position: 0% 50%;
  }
}

/* Media query for responsiveness */
@media (max-width: 935px) {
  body {
    padding: 10px 20px;
  }

  header h1,
  header p {
    font-size: 2rem;
  }

  input {
    width: 80%;
  }

  .card-grid {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 20px;
  }

  .card {
    width: 100%;
  }

  .card img {
    height: 150px;
  }
}
