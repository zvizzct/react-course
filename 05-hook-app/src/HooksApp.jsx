import { useReducer } from 'react';

export const HooksApp = () => {
  return <div>HooksApp</div>;
};
