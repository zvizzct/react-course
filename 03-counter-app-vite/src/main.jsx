import React from 'react'
import ReactDOM from 'react-dom/client'
import { CounterApp } from './CounterApp'
import { FirstApp } from './FirstApp'

import './style.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <FirstApp title={"lol"} /> */}
    <CounterApp value={20} />
  </React.StrictMode >,
)
