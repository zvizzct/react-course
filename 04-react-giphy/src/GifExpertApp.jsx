import { useState } from 'react';
import { AddCategory, GifGrid, Header, GifHistory } from './components';
export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['The Office']);
  const [history, setHistory] = useState([]);

  const onSelectCategory = (selectedCat) => {
    setCategories([selectedCat]);
  };
  const onAddCategory = (newCat) => {
    if (categories.includes(newCat)) return;
    setCategories((prevCategories) => [newCat]);
    if (history && history.includes(newCat)) return;
    setHistory((prevHistory) => [newCat, ...prevHistory]);
  };

  return (
    <div className="container">
      <Header />
      <AddCategory onNewCategory={onAddCategory} />
      <GifHistory history={history} onSelectCategory={onSelectCategory} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </div>
  );
};
