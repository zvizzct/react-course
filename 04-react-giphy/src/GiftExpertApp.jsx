import { useState } from 'react';
import { AddCategory, GifGrid, Header, GifHistory } from './components';
export const GiftExpertApp = () => {
  const [categories, setCategories] = useState([]);
  const [history, setHistory] = useState([]);

  const onSelectCategory = (selectedCat) => {
    setCategories([selectedCat]);
  };
  const onAddCategory = (newCat) => {
    if (categories.includes(newCat)) return;
    if (history && history.includes(newCat)) return;
    setCategories((prevCategories) => [newCat]);
    setHistory((prevHistory) => [newCat, ...prevHistory]);
    console.log(newCat, categories, history);
  };

  return (
    <div className="container">
      <Header />
      <AddCategory onNewCategory={onAddCategory} />
      <GifHistory history={history} onSelectCategory={onSelectCategory} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </div>
  );
};
