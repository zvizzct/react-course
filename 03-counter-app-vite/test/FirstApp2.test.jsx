import { render, screen } from '@testing-library/react'
import { FirstApp } from '../src/FirstApp'


describe('Pruebas en <FirstApp/>', () => {

    const title = 'hola soy goku'
    const subTitle = 'Soy un subtitulo'

    test('debe de hacer match con el snapshot', () => {
        const { container } = render(<FirstApp title={title} />)
        expect(container).toMatchSnapshot()
    })

    test('debe de mostrar el mensaje "hola soy Goku"', () => {
        render(<FirstApp title={title} />)
        expect(screen.getByText(title)).toBeTruthy()
        // expect(getByText(title)).toBeInTheDocument()
    })

    test('debe de mostrar el titulo en h1', () => {
        render(<FirstApp title={title} />)
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)
    })
    test('debe de mostrar el subtitulo enviado por props', () => {
        render(<FirstApp title={title} subTitle={subTitle} />)
        expect(screen.getByText(subTitle)).toBeTruthy()
    })
})