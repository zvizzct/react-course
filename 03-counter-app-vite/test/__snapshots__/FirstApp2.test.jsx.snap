// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    hola soy goku
  </h1>
  <p>
    No hay sub
  </p>
  <p>
    No hay nombre
  </p>
</div>
`;
